<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>README &mdash; Microbial Ecology Protocols 0.2 documentation</title>
    
    <link rel="stylesheet" href="../_static/labibi.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '0.2',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <link rel="top" title="Microbial Ecology Protocols 0.2 documentation" href="../index.html" />


<style type="text/css">
<!-- github-based editing disabled; set github_base_account
     and github_project -->
#editor-trap.toggled {
  display: none;
}
</style>



<!-- Google Analytics JS is disabled; set google_analytics_id -->


  </head>
  <body role="document">
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="nav-item nav-item-0"><a href="../index.html">Microbial Ecology Protocols 0.2 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
          <div class="body" role="main">
            
  
  <p>Set up working directory:</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nb">cd</span> /usr/local/notebooks
mkdir -p ./workdir
</pre></div>
</div>
<p>Check seqfile files to process in data directory (make sure you still remember the data directory):</p>
<div class="highlight-bash"><div class="highlight"><pre>ls ./data/test/data
</pre></div>
</div>
<div class="section" id="readme">
<h1>README<a class="headerlink" href="#readme" title="Permalink to this headline">Â¶</a></h1>
<p><a href="#id1"><span class="problematic" id="id2">**</span></a>This part of pipeline search for the SSU rRNA gene fragments, classify them, and extract reads aligned specific region. It is also heavy lifting part of the whole pipeline (more cpu will help).</p>
<p>This part works with one seqfile a time. You just need to change the &#8220;Seqfile&#8221; and maybe other parameters in the two cells bellow.**</p>
<p>If your computer has <strong>many processors</strong>, there are two ways to make use of the resource:</p>
<ol class="arabic simple">
<li>Set &#8220;Cpu&#8221; higher number.</li>
<li>make more copies of this notebook (click &#8220;File&#8221; then &#8220;Make a copy&#8221; in
menu bar), so you can run the step on multiple files at the same
time.</li>
</ol>
<p>(Again we assume the &#8220;Seqfile&#8221; is quality trimmed.)</p>
<p><a href="#id3"><span class="problematic" id="id4">**</span></a>Here we will process one file at a time; set the &#8220;Seqfile&#8221; variable to the seqfile name to be be processed</p>
<p>First part of seqfile basename (separated by &#8221;.&#8221;) will be the label of this sample, so named it properly.**</p>
<p>e.g. for &#8220;/usr/local/notebooks/data/test/data/1c.fa&#8221;, &#8220;1c&#8221; will the
label of this sample.</p>
<p>Set the Seqfile variable to the sequence file to process:</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">Seqfile</span><span class="o">=</span><span class="s1">&#39;./data/test/data/2c.fa&#39;</span>
</pre></div>
</div>
<p>Other parameters to set:</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">Cpu</span><span class="o">=</span><span class="s1">&#39;2&#39;</span>   <span class="c"># number of maxixum threads for search and alignment</span>
<span class="nv">Hmm</span><span class="o">=</span><span class="s1">&#39;./data/SSUsearch_db/Hmm.ssu.hmm&#39;</span>   <span class="c"># hmm model for ssu</span>
<span class="nv">Gene</span><span class="o">=</span><span class="s1">&#39;ssu&#39;</span>
<span class="nv">Script_dir</span><span class="o">=</span><span class="s1">&#39;./SSUsearch/scripts&#39;</span>
<span class="nv">Gene_model_org</span><span class="o">=</span><span class="s1">&#39;./data/SSUsearch_db/Gene_model_org.16s_ecoli_J01695.fasta&#39;</span>
<span class="nv">Ali_template</span><span class="o">=</span><span class="s1">&#39;./data/SSUsearch_db/Ali_template.silva_ssu.fasta&#39;</span>

<span class="nv">Start</span><span class="o">=</span><span class="s1">&#39;577&#39;</span>  <span class="c">#pick regions for de novo clustering</span>
<span class="nv">End</span><span class="o">=</span><span class="s1">&#39;727&#39;</span>
<span class="nv">Len_cutoff</span><span class="o">=</span><span class="s1">&#39;100&#39;</span> <span class="c"># min length for reads picked for the region</span>

<span class="nv">Gene_tax</span><span class="o">=</span><span class="s1">&#39;./data/SSUsearch_db/Gene_tax.silva_taxa_family.tax&#39;</span> <span class="c"># silva 108 ref</span>
<span class="nv">Gene_db</span><span class="o">=</span><span class="s1">&#39;./data/SSUsearch_db/Gene_db.silva_108_rep_set.fasta&#39;</span>

<span class="nv">Gene_tax_cc</span><span class="o">=</span><span class="s1">&#39;./data/SSUsearch_db/Gene_tax_cc.greengene_97_otus.tax&#39;</span> <span class="c"># greengene 2012.10 ref for copy correction</span>
<span class="nv">Gene_db_cc</span><span class="o">=</span><span class="s1">&#39;./data/SSUsearch_db/Gene_db_cc.greengene_97_otus.fasta&#39;</span>

<span class="c">###</span>
<span class="c">### do not need to change lines below</span>
<span class="c">###</span>

<span class="c">### first part of file basename will the label of this sample</span>
<span class="nv">Filename</span><span class="o">=</span><span class="k">$(</span>basename<span class="o">(</span><span class="si">${</span><span class="nv">Seqfile</span><span class="si">}</span><span class="k">)</span><span class="o">)</span>
<span class="nv">Tag</span><span class="o">=</span><span class="si">${</span><span class="nv">Filename</span><span class="p">%%.*</span><span class="si">}</span>

<span class="c">### make absolute path</span>
<span class="nb">export </span><span class="nv">Hmm</span><span class="o">=</span><span class="k">$(</span>readlink -f <span class="si">${</span><span class="nv">Hmm</span><span class="si">}</span><span class="k">)</span>
<span class="nb">export </span><span class="nv">Seqfile</span><span class="o">=</span><span class="k">$(</span>readlink -f <span class="si">${</span><span class="nv">Seqfile</span><span class="si">}</span><span class="k">)</span>
<span class="nb">export </span><span class="nv">Script_dir</span><span class="o">=</span><span class="k">$(</span>readlink -f <span class="si">${</span><span class="nv">Script_dir</span><span class="si">}</span><span class="k">)</span>
<span class="nb">export </span><span class="nv">Gene_model_org</span><span class="o">=</span><span class="k">$(</span>readlink -f <span class="si">${</span><span class="nv">Gene_model_org</span><span class="si">}</span><span class="k">)</span>
<span class="nb">export </span><span class="nv">Ali_template</span><span class="o">=</span><span class="k">$(</span>readlink -f <span class="si">${</span><span class="nv">Ali_template</span><span class="si">}</span><span class="k">)</span>
<span class="nb">export </span><span class="nv">Gene_tax</span><span class="o">=</span><span class="k">$(</span>readlink -f <span class="si">${</span><span class="nv">Gene_tax</span><span class="si">}</span><span class="k">)</span>
<span class="nb">export </span><span class="nv">Gene_db</span><span class="o">=</span><span class="k">$(</span>readlink -f <span class="si">${</span><span class="nv">Gene_db</span><span class="si">}</span><span class="k">)</span>
<span class="nb">export </span><span class="nv">Gene_tax_cc</span><span class="o">=</span><span class="k">$(</span>readlink -f <span class="si">${</span><span class="nv">Gene_tax_cc</span><span class="si">}</span><span class="k">)</span>
<span class="nb">export </span><span class="nv">Gene_db_cc</span><span class="o">=</span><span class="k">$(</span>readlink -f <span class="si">${</span><span class="nv">Gene_db_cc</span><span class="si">}</span><span class="k">)</span>
</pre></div>
</div>
<p>Double check key variables:</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nb">echo</span> <span class="s2">&quot;*** make sure: parameters are right&quot;</span>
<span class="nb">echo</span> <span class="s2">&quot;Seqfile: </span><span class="nv">$Seqfile</span><span class="s2">\nCpu: </span><span class="nv">$Cpu</span><span class="s2">\nFilename: </span><span class="nv">$Filename</span><span class="s2">\nTag: </span><span class="nv">$Tag</span><span class="s2">&quot;</span>
</pre></div>
</div>
<p>Do the hmmsearch (<strong>heavy weight lifting part</strong>):</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nb">cd </span>workdir
mkdir -p <span class="nv">$Tag</span>.ssu.out

<span class="c">### start hmmsearch</span>
<span class="nb">echo</span> <span class="s2">&quot;*** hmmsearch starting&quot;</span>
<span class="nb">time </span>hmmsearch --incE <span class="m">10</span> --incdomE <span class="m">10</span> --cpu <span class="nv">$Cpu</span> <span class="se">\</span>
  --domtblout <span class="nv">$Tag</span>.ssu.out/<span class="nv">$Tag</span>.qc.<span class="nv">$Gene</span>.hmmdomtblout <span class="se">\</span>
  -o /dev/null -A <span class="nv">$Tag</span>.ssu.out/<span class="nv">$Tag</span>.qc.<span class="nv">$Gene</span>.sto <span class="se">\</span>
  <span class="nv">$Hmm</span> <span class="nv">$Seqfile</span>
<span class="nb">echo</span> <span class="s2">&quot;*** hmmsearch finished&quot;</span>
</pre></div>
</div>
<p>Parsing the output:</p>
<div class="highlight-bash"><div class="highlight"><pre>python <span class="nv">$Script_dir</span>/get-seq-from-hmmout.py <span class="se">\</span>
    <span class="nv">$Tag</span>.ssu.out/<span class="nv">$Tag</span>.qc.<span class="nv">$Gene</span>.hmmdomtblout <span class="se">\</span>
    <span class="nv">$Tag</span>.ssu.out/<span class="nv">$Tag</span>.qc.<span class="nv">$Gene</span>.sto <span class="se">\</span>
    <span class="nv">$Tag</span>.ssu.out/<span class="nv">$Tag</span>.qc.<span class="nv">$Gene</span>
</pre></div>
</div>
<p>Pass hits to mothur aligner:</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nb">echo</span> <span class="s2">&quot;*** Starting mothur align&quot;</span>
cat  <span class="nv">$Gene_model_org</span> <span class="nv">$Tag</span>.ssu.out/<span class="nv">$Tag</span>.qc.<span class="nv">$Gene</span> &gt; <span class="nv">$Tag</span>.ssu.out/<span class="nv">$Tag</span>.qc.<span class="nv">$Gene</span>.RFadded

<span class="c"># mothur does not allow tab between its flags, thus no indents here</span>
<span class="nb">time </span>mothur <span class="s2">&quot;#align.seqs(candidate=</span><span class="nv">$Tag</span><span class="s2">.ssu.out/</span><span class="nv">$Tag</span><span class="s2">.qc.</span><span class="nv">$Gene</span><span class="s2">.RFadded, template=</span><span class="nv">$Ali_template</span><span class="s2">, threshold=0.5, flip=t, processors=</span><span class="nv">$Cpu</span><span class="s2">)&quot;</span>

rm -f mothur.*.logfile
</pre></div>
</div>
<p>Get aligned seqs that have &gt; 50% matched to references:</p>
<div class="highlight-bash"><div class="highlight"><pre>python <span class="nv">$Script_dir</span>/mothur-align-report-parser-cutoff.py <span class="se">\</span>
    <span class="nv">$Tag</span>.ssu.out/<span class="nv">$Tag</span>.qc.<span class="nv">$Gene</span>.align.report <span class="se">\</span>
    <span class="nv">$Tag</span>.ssu.out/<span class="nv">$Tag</span>.qc.<span class="nv">$Gene</span>.align <span class="se">\</span>
    <span class="nv">$Tag</span>.ssu.out/<span class="nv">$Tag</span>.qc.<span class="nv">$Gene</span>.align.filter <span class="se">\</span>
    0.5
</pre></div>
</div>
<p>Get the unaligned fasta file:</p>
<div class="highlight-bash"><div class="highlight"><pre>python <span class="nv">$Script_dir</span>/remove-gap.py <span class="nv">$Tag</span>.ssu.out/<span class="nv">$Tag</span>.qc.<span class="nv">$Gene</span>.align.filter <span class="nv">$Tag</span>.ssu.out/<span class="nv">$Tag</span>.qc.<span class="nv">$Gene</span>.align.filter.fa
</pre></div>
</div>
<p><strong>Search is done here (the computational intensive part). Hooray! There are two useful output files:</strong></p>
<ul class="simple">
<li>$Tag.ssu.out/$Tag.qc.$Gene.align.filter:
aligned SSU rRNA gene fragments</li>
<li>$Tag.ssu.out/$Tag.qc.$Gene.align.filter.fa:
unaligned SSU rRNA gene fragments</li>
</ul>
<p>Extract the reads mapped 150bp region in V4 (577-727 in *E.coli* SSU rRNA gene position) for unsupervised clustering:</p>
<div class="highlight-bash"><div class="highlight"><pre>python <span class="nv">$Script_dir</span>/region-cut.py <span class="nv">$Tag</span>.ssu.out/<span class="nv">$Tag</span>.qc.<span class="nv">$Gene</span>.align.filter <span class="nv">$Start</span> <span class="nv">$End</span> <span class="nv">$Len_cutoff</span>

mv <span class="nv">$Tag</span>.ssu.out/<span class="nv">$Tag</span>.qc.<span class="nv">$Gene</span>.align.filter.<span class="s2">&quot;</span><span class="nv">$Start</span><span class="s2">&quot;</span>to<span class="s2">&quot;</span><span class="nv">$End</span><span class="s2">&quot;</span>.cut.lenscreen <span class="nv">$Tag</span>.ssu.out/<span class="nv">$Tag</span>.forclust
</pre></div>
</div>
<p>Classify SSU rRNA gene seqs using SILVA:</p>
<div class="highlight-bash"><div class="highlight"><pre>rm -f <span class="nv">$Tag</span>.ssu.out/<span class="nv">$Tag</span>.qc.<span class="nv">$Gene</span>.align.filter.*.wang.taxonomy
mothur <span class="s2">&quot;#classify.seqs(fasta=</span><span class="nv">$Tag</span><span class="s2">.ssu.out/</span><span class="nv">$Tag</span><span class="s2">.qc.</span><span class="nv">$Gene</span><span class="s2">.align.filter.fa, template=</span><span class="nv">$Gene_db</span><span class="s2">, taxonomy=</span><span class="nv">$Gene_tax</span><span class="s2">, cutoff=50, processors=</span><span class="nv">$Cpu</span><span class="s2">)&quot;</span>
mv <span class="nv">$Tag</span>.ssu.out/<span class="nv">$Tag</span>.qc.<span class="nv">$Gene</span>.align.filter.*.wang.taxonomy <span class="se">\</span>
    <span class="nv">$Tag</span>.ssu.out/<span class="nv">$Tag</span>.qc.<span class="nv">$Gene</span>.align.filter.wang.silva.taxonomy
</pre></div>
</div>
<p>Get the *.taxonomy file has taxon for each SSU rRNA fragment sequence id. We can get the count for each taxon:</p>
<div class="highlight-bash"><div class="highlight"><pre>python <span class="nv">$Script_dir</span>/count-taxon.py <span class="se">\</span>
    <span class="nv">$Tag</span>.ssu.out/<span class="nv">$Tag</span>.qc.<span class="nv">$Gene</span>.align.filter.wang.silva.taxonomy <span class="se">\</span>
    <span class="nv">$Tag</span>.ssu.out/<span class="nv">$Tag</span>.qc.<span class="nv">$Gene</span>.align.filter.wang.silva.taxonomy.count
rm -f mothur.*.logfile
</pre></div>
</div>
<p>Classify SSU rRNA gene seqs with Greengene for copy correction later:</p>
<div class="highlight-bash"><div class="highlight"><pre>rm -f <span class="nv">$Tag</span>.ssu.out/<span class="nv">$Tag</span>.qc.<span class="nv">$Gene</span>.align.filter.*.wang.taxonomy
mothur <span class="s2">&quot;#classify.seqs(fasta=</span><span class="nv">$Tag</span><span class="s2">.ssu.out/</span><span class="nv">$Tag</span><span class="s2">.qc.</span><span class="nv">$Gene</span><span class="s2">.align.filter.fa, template=</span><span class="nv">$Gene_db_cc</span><span class="s2">, taxonomy=</span><span class="nv">$Gene_tax_cc</span><span class="s2">, cutoff=50, processors=</span><span class="nv">$Cpu</span><span class="s2">)&quot;</span>
mv <span class="nv">$Tag</span>.ssu.out/<span class="nv">$Tag</span>.qc.<span class="nv">$Gene</span>.align.filter.*.wang.taxonomy <span class="se">\</span>
    <span class="nv">$Tag</span>.ssu.out/<span class="nv">$Tag</span>.qc.<span class="nv">$Gene</span>.align.filter.wang.gg.taxonomy
</pre></div>
</div>
<p>Count the taxon:</p>
<div class="highlight-bash"><div class="highlight"><pre>python <span class="nv">$Script_dir</span>/count-taxon.py <span class="se">\</span>
    <span class="nv">$Tag</span>.ssu.out/<span class="nv">$Tag</span>.qc.<span class="nv">$Gene</span>.align.filter.wang.gg.taxonomy <span class="se">\</span>
    <span class="nv">$Tag</span>.ssu.out/<span class="nv">$Tag</span>.qc.<span class="nv">$Gene</span>.align.filter.wang.gg.taxonomy.count
rm -f mothur.*.logfile
</pre></div>
</div>
<p>Check the output directory:</p>
<div class="highlight-bash"><div class="highlight"><pre>ls <span class="nv">$Tag</span>.ssu.out
</pre></div>
</div>
<p>Here is the a list of output files:
.. parsed-literal:</p>
<div class="highlight-bash"><div class="highlight"><pre>1c.577to727
1c.cut
1c.forclust
1c.qc.ssu
1c.qc.ssu.align
1c.qc.ssu.align.filter
1c.qc.ssu.align.filter.577to727.cut
1c.qc.ssu.align.filter.577to727.cut.lenscreen.fa
1c.qc.ssu.align.filter.fa
1c.qc.ssu.align.filter.greengene_97_otus.wang.tax.summary
1c.qc.ssu.align.filter.silva_taxa_family.wang.tax.summary
1c.qc.ssu.align.filter.wang.gg.taxonomy
1c.qc.ssu.align.filter.wang.gg.taxonomy.count
1c.qc.ssu.align.filter.wang.silva.taxonomy
1c.qc.ssu.align.filter.wang.silva.taxonomy.count
1c.qc.ssu.align.report
1c.qc.ssu.hmmdomtblout
1c.qc.ssu.hmmdomtblout.parsedToDictWithScore.pickle
1c.qc.ssu.hmmtblout
1c.qc.ssu.RFadded
1c.qc.ssu.sto
</pre></div>
</div>
<p><strong>This part of pipeline (working with one sequence file) finishes here. Next we will combine samples for community analysis (see unsupervised analysis).</strong></p>
<p><strong>Following are files useful for community analysis</strong>:</p>
<ul class="simple">
<li>1c.577to727: aligned fasta file of seqs mapped to target region for
de novo clustering</li>
<li>1c.qc.ssu.align.filter: aligned fasta file of all SSU rRNA gene
fragments</li>
<li>1c.qc.ssu.align.filter.wang.gg.taxonomy: Greengene taxonomy (for copy
correction)</li>
<li>1c.qc.ssu.align.filter.wang.silva.taxonomy: SILVA taxonomy</li>
</ul>
</div>




<!-- disqus commenting disabled; set disqus_shortname -->

          </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li class="nav-item nav-item-0"><a href="../index.html">Microbial Ecology Protocols 0.2 documentation</a> &raquo;</li> 
      </ul>
    </div>

    <div class="footer" role="contentinfo">
        &copy; Copyright 2015, Jiarong Guo.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.3.1.
    </div>




<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', ]);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>


  </body>
</html>