<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Unsupervised analysis &mdash; Microbial Ecology Protocols 0.2 documentation</title>
    
    <link rel="stylesheet" href="../_static/labibi.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '0.2',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <link rel="top" title="Microbial Ecology Protocols 0.2 documentation" href="../index.html" />


<style type="text/css">
<!-- github-based editing disabled; set github_base_account
     and github_project -->
#editor-trap.toggled {
  display: none;
}
</style>



<!-- Google Analytics JS is disabled; set google_analytics_id -->


  </head>
  <body role="document">
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="nav-item nav-item-0"><a href="../index.html">Microbial Ecology Protocols 0.2 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
          <div class="body" role="main">
            
  
  <div class="section" id="unsupervised-analysis">
<h1>Unsupervised analysis<a class="headerlink" href="#unsupervised-analysis" title="Permalink to this headline">Â¶</a></h1>
<p>Set another directory for unsupervised analysis:</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nb">cd</span> /usr/local/notebooks
mkdir -p ./workdir/clust
</pre></div>
</div>
<p>Set parameters:</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">Prefix</span><span class="o">=</span><span class="s1">&#39;SS&#39;</span>    <span class="c"># name for the analysis run</span>
<span class="nv">Script_dir</span><span class="o">=</span><span class="s1">&#39;./SSUsearch/scripts&#39;</span>
<span class="nv">Wkdir</span><span class="o">=</span><span class="s1">&#39;./workdir&#39;</span>
<span class="nv">Mcclust_jar</span><span class="o">=</span><span class="s1">&#39;./external_tools/Clustering/dist/Clustering.jar&#39;</span>
<span class="nv">Java_xmx</span><span class="o">=</span><span class="s1">&#39;10g&#39;</span>
<span class="nv">Java_gc_threads</span><span class="o">=</span><span class="s1">&#39;2&#39;</span>
<span class="nv">Otu_dist_cutoff</span><span class="o">=</span><span class="s1">&#39;0.05&#39;</span>
<span class="nv">Design</span><span class="o">=</span><span class="s1">&#39;./data/test/SS.design&#39;</span>

<span class="c"># get absolute path</span>
<span class="nv">Script_dir</span><span class="o">=</span><span class="k">$(</span>readlink -f <span class="si">${</span><span class="nv">Script_dir</span><span class="si">}</span><span class="k">)</span>
<span class="nv">Wkdir</span><span class="o">=</span><span class="k">$(</span>readlink -f <span class="si">${</span><span class="nv">Wkdir</span><span class="si">}</span><span class="k">)</span>
<span class="nv">Mcclust_jar</span><span class="o">=</span><span class="k">$(</span>readlink -f <span class="si">${</span><span class="nv">Mcclust_jar</span><span class="si">}</span><span class="k">)</span>
<span class="nv">Design</span><span class="o">=</span><span class="k">$(</span>readlink -f <span class="si">${</span><span class="nv">Design</span><span class="si">}</span><span class="k">)</span>
</pre></div>
</div>
<p>Go the clust direcory:</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nb">cd</span> ./workdir/clust
</pre></div>
</div>
<p>Combine all the aligned sequences from each sample:</p>
<div class="highlight-bash"><div class="highlight"><pre>cat <span class="nv">$Wkdir</span>/*.ssu.out/*.forclust &gt; combined_seqs.afa
</pre></div>
</div>
<p>Make group file for mcclust and mothur. First part of the file basename will be the group label, e.g. file &#8220;aa.bb.cc&#8221; will have &#8220;aa&#8221; as group label:</p>
<div class="highlight-bash"><div class="highlight"><pre>python <span class="nv">$Script_dir</span>/make-groupfile.py <span class="nv">$Prefix</span>.groups <span class="nv">$Wkdir</span>/*.ssu.out/*.forclust
</pre></div>
</div>
<p>Dereplication with mcclust:</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nb">echo</span> <span class="s2">&quot;*** Starting mcclust derep..&quot;</span>
<span class="nb">time </span>java -Xmx<span class="nv">$Java_xmx</span> -XX:+UseParallelOldGC -XX:ParallelGCThreads<span class="o">=</span><span class="nv">$Java_gc_threads</span> -jar <span class="nv">$Mcclust_jar</span> derep -a -o derep.fasta temp.mcclust.names temp.txt combined_seqs.afa
rm temp.txt
</pre></div>
</div>
<p>Precluster to collapse sequences with 1 base difference (most likely caused by sequencing errors):</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="c">#Convert mcclust names to mothur names</span>
python <span class="nv">$Script_dir</span>/mcclust2mothur_names_file.py temp.mcclust.names temp.mothur.names

<span class="nb">echo</span> <span class="s2">&quot;***starting preclust..&quot;</span>
<span class="c"># output: derep.precluster.fasta, derep.precluster.names</span>
mothur <span class="s2">&quot;#pre.cluster(fasta=derep.fasta, diffs=1, name=temp.mothur.names)&quot;</span>

<span class="c">#Convert names back to mcclust</span>
python <span class="nv">$Script_dir</span>/mothur2mcclust_names_file.py derep.precluster.names <span class="nv">$Prefix</span>.names
</pre></div>
</div>
<p>Make distance matrix:</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nb">time </span>java -Xmx<span class="nv">$Java_xmx</span> -XX:+UseParallelOldGC -XX:ParallelGCThreads<span class="o">=</span><span class="nv">$Java_gc_threads</span> -jar <span class="nv">$Mcclust_jar</span> dmatrix -l <span class="m">25</span> -o matrix.bin -i <span class="nv">$Prefix</span>.names -I derep.precluster.fasta
</pre></div>
</div>
<p>Clustering with mcclust:</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nb">time </span>java -Xmx<span class="nv">$Java_xmx</span> -XX:+UseParallelOldGC -XX:ParallelGCThreads<span class="o">=</span><span class="nv">$Java_gc_threads</span> -jar <span class="nv">$Mcclust_jar</span> cluster -m upgma -i <span class="nv">$Prefix</span>.names -s <span class="nv">$Prefix</span>.groups -o complete.clust -d matrix.bin
</pre></div>
</div>
<p>Convert clust results to mothur list:</p>
<div class="highlight-bash"><div class="highlight"><pre>python <span class="nv">$Script_dir</span>/mcclust2mothur-list-cutoff.py complete.clust <span class="nv">$Prefix</span>.list <span class="nv">$Otu_dist_cutoff</span>
</pre></div>
</div>
<p>Remove &#8221;:&#8221; in sequence names (mothur automatically converts &#8221;:&#8221; to &#8220;_&#8221;):</p>
<div class="highlight-bash"><div class="highlight"><pre>sed -i <span class="s1">&#39;s/:/_/g&#39;</span> <span class="nv">$Prefix</span>.names <span class="nv">$Prefix</span>.groups <span class="nv">$Prefix</span>.list
<span class="nb">echo</span> <span class="s2">&quot;*** Replace &#39;:&#39; with &#39;_&#39; in seq names (original illumina name has &#39;:&#39; in them)&quot;</span>
</pre></div>
</div>
<p>Get representitives of each OTU at OTU distance cutoff defined at top of this page:</p>
<div class="highlight-bash"><div class="highlight"><pre>java -jar <span class="nv">$Mcclust_jar</span> rep-seqs -c -l -s complete.clust <span class="nv">$Otu_dist_cutoff</span> combined_seqs.afa
mv complete.clust_rep_seqs.fasta otu_rep_align.fa
</pre></div>
</div>
<dl class="docutils">
<dt>Get OTU table::</dt>
<dd>mothur &#8220;#make.shared(list=$Prefix.list, group=$Prefix.groups, label=$Otu_dist_cutoff);&#8221;</dd>
</dl>
<p>Get consensus taxonomy for each OTU:</p>
<div class="highlight-bash"><div class="highlight"><pre>cat <span class="nv">$Wkdir</span>/*.ssu.out/*.silva.taxonomy &gt; <span class="nv">$Prefix</span>.taxonomy
mothur <span class="s2">&quot;#classify.otu(list=</span><span class="nv">$Prefix</span><span class="s2">.list, taxonomy=</span><span class="nv">$Prefix</span><span class="s2">.taxonomy, label=</span><span class="nv">$Otu_dist_cutoff</span><span class="s2">)&quot;</span>
</pre></div>
</div>
<p>Make biom file:</p>
<div class="highlight-bash"><div class="highlight"><pre>mothur <span class="s2">&quot;#make.biom(shared=</span><span class="nv">$Prefix</span><span class="s2">.shared, constaxonomy=</span><span class="nv">$Prefix</span><span class="s2">.</span><span class="nv">$Otu_dist_cutoff</span><span class="s2">.cons.taxonomy)&quot;</span>
mv <span class="nv">$Prefix</span>.<span class="nv">$Otu_dist_cutoff</span>.biom <span class="nv">$Prefix</span>.biom

<span class="c"># clean up tempfiles</span>
rm -f mothur.*.logfile *rabund <span class="nb">complete</span>* derep.fasta matrix.bin nonoverlapping.bin temp.*
</pre></div>
</div>
<p><strong>With SS.groups, SS.names and SS.list</strong>, most diversity analysis can be done by mothur. You can look at <a class="reference external" href="http://www.mothur.org/wiki/454_SOP">mothur wiki</a> for details (Do not forgot to do even sampling before beta-diversity analysis).</p>
<p><strong>SS.biom file can used in most tools. (qiime, rdp, and phyloseq)</strong></p>
<p>Since the purpose of this tutorial is to show our new pipeline, we will skip details of community analysis with mothur. Following are some common commands in mothur:</p>
<div class="highlight-bash"><div class="highlight"><pre>mothur <span class="s2">&quot;#make.shared(biom=</span><span class="nv">$Prefix</span><span class="s2">.biom); sub.sample(shared=</span><span class="nv">$Prefix</span><span class="s2">.shared); summary.single(calc=nseqs-coverage-sobs-chao-shannon-invsimpson); dist.shared(calc=braycurtis); pcoa(phylip=</span><span class="nv">$Prefix</span><span class="s2">.userLabel.subsample.braycurtis.userLabel.lt.dist); nmds(phylip=</span><span class="nv">$Prefix</span><span class="s2">.userLabel.subsample.braycurtis.userLabel.lt.dist); amova(phylip=</span><span class="nv">$Prefix</span><span class="s2">.userLabel.subsample.braycurtis.userLabel.lt.dist, design=</span><span class="nv">$Design</span><span class="s2">); tree.shared(calc=braycurtis); unifrac.weighted(tree=</span><span class="nv">$Prefix</span><span class="s2">.userLabel.subsample.braycurtis.userLabel.tre, group=</span><span class="nv">$Design</span><span class="s2">, random=T)&quot;</span>
rm -f mothur.*.logfile<span class="p">;</span>
rm -f *.rabund
</pre></div>
</div>
<p>Some simple visualization:</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="c"># alpha diveristy index</span>
python <span class="nv">$Script_dir</span>/plot-diversity-index.py <span class="s2">&quot;userLabel&quot;</span> <span class="s2">&quot;chao,shannon,invsimpson&quot;</span> <span class="s2">&quot;c,d&quot;</span> <span class="s2">&quot;SS.userLabel.subsample.groups.summary&quot;</span> <span class="s2">&quot;test&quot;</span> <span class="s2">&quot;test.alpha&quot;</span>

<span class="c"># taxon distribution</span>
python <span class="nv">$Script_dir</span>/plot-taxa-count.py <span class="m">2</span> test.taxa.dist ../*.ssu.out/*.silva.taxonomy.count

<span class="c"># ordination</span>
python <span class="nv">$Script_dir</span>/plot-pcoa.py  SS.userLabel.subsample.braycurtis.userLabel.lt.pcoa.axes  SS.userLabel.subsample.braycurtis.userLabel.lt.pcoa.loadings  test.beta.pcoa
</pre></div>
</div>
</div>




<!-- disqus commenting disabled; set disqus_shortname -->

          </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li class="nav-item nav-item-0"><a href="../index.html">Microbial Ecology Protocols 0.2 documentation</a> &raquo;</li> 
      </ul>
    </div>

    <div class="footer" role="contentinfo">
        &copy; Copyright 2015, Jiarong Guo.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.3.1.
    </div>




<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', ]);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>


  </body>
</html>